graphql/resolvers.go

```diff
@@ -1098,6 +1099,64 @@ func (r *mutationResolver) SetPatchPriority(ctx context.Context, patchID string,
 	return &patchID, nil
 }
 
+func (r *mutationResolver) EnqueuePatch(ctx context.Context, patchID string) (*restModel.APIPatch, error) {
+	user := route.MustHaveUser(ctx)
+	hasPermission, err := r.hasEnqueuePatchPermission(user, patchID)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("error getting permissions: %s", err.Error()))
+	}
+	if !hasPermission {
+		return nil, Forbidden.Send(ctx, "can't enqueue another user's patch")
```

Comment:
Should a check like this also exist in the merge_patch route in case someone uses the API directly? (Potentially in the CanEnqueueToCommitQueue function, also apologies if this is already handled somewhere and I didn't find it!)

Reply:
The merge_patch route passes through the [commit queue middleware](https://github.com/evergreen-ci/evergreen/blob/60ba9e0c3c9ddbb9f18e207510387597fedb8bdd/rest/route/middleware.go#L274) which checks the user owns the patch.